### ISSUE TITLE
Compose key not working

### ISSUE THREAD

**Issue by guyzmo** [equalsraf/neovim-qt#181]
**OS**: ArchLinux
**Version**: git master (taken through the AUR install)
**Expected**: When compose keys are enabled in Xorg, <kbd>Compose</kbd>+<kbd>:</kbd>+<kbd>)</kbd> should display `☺`… I get `:)` instead.

it looks like the key setup as <kbd>Compose</kbd> key is falling through. In my configuration, I setup the right <kbd>Alt</kbd> as Xorg's <kbd>Multi_key</kbd> which is set up to be the compose key.

**Comment by cjquines** [comment]
hello, i am having the same issue; is there anything i can do to help fix this?

**Comment by cjquines** [comment]
right. after i do the sequence of presses <kbd>Compose</kbd> <kbd>a</kbd> <kbd>'</kbd> i get

```
QKeyEvent ev: QKeyEvent(KeyPress, Key_AltGr, GroupSwitchModifier)
   ""
QKeyEvent ev: QKeyEvent(KeyPress, Key_Apostrophe, text="'")
   "'"
```

so there's no event corresponding to <kbd>a</kbd>—it gets eaten, for some reason?

on the other hand, if i press <kbd>Compose</kbd> <kbd>Compose</kbd>, it picks up the second one as `Key_Multi_key`:

```
QKeyEvent ev: QKeyEvent(KeyPress, Key_AltGr, GroupSwitchModifier)
   ""
QKeyEvent ev: QKeyEvent(KeyPress, Key_Multi_key)
   "ᄠ"
```

**Comment by jgehrig** [comment]
@cjquines 

Some additional context would be helpful. I'm not familiar with the "Compose" key. Is this for inputting CJK characters, emoji, both, or something else?

We'll probably get a few key events that we should ignore... As of today, the input layer is stateless. I'd like to keep it this way if possible, and avoid adding behavior that is dependent on past `QKeyEvent` sequences.

If I understand correctly:
1. The first sequence is what you expect to to insert an emoji. I agree there is nothing we can do with this input. Qt bug?
2. The second sequence looks promising, we can probably parse that input correctly.However the double "Compose" key is odd.  Is `ᄠ` what you expect? 

It is also possible that Qt does not handle this scenario correctly, and there is a bug (or not-handled-scenario) upstream. What version of Qt are you built against? Check `nvim-qt --version`.

**Comment by cjquines** [comment]
the compose key works as described on, say, [this wikipedia page](https://en.wikipedia.org/wiki/Compose_key). in most applications, when i input the sequence <kbd>Compose</kbd><kbd>a</kbd><kbd>'</kbd> it produces the character `á`. from what i understand this is a default x11 combination.

on the other hand, i don't have anything special mapped to <kbd>Compose</kbd><kbd>Compose</kbd>.

upon further testing, this only seems to be the case for compose combinations that come from default with x11. in my system these are in `/usr/share/X11/locale/en_US.UTF-8/Compose`. compose sequences that i've defined in an `~/.XCompose` file *do* work. for example, i defined <kbd>Compose</kbd> <kbd>G</kbd> <kbd>a</kbd> to map to `α`. when i input this, the only event appears is

```
QKeyEvent ev: QKeyEvent(KeyPress, Key_AltGr, GroupSwitchModifier)
   ""
```

and yet it *does* produce the desired result of `α`. i suspect that this is actually a bug on the level of qt, then.

running `./nvim-qt --version` on the nvim-qt that i build gives `Qt Version: 5.15.2`.
